ops/model.py

```diff
@@ -638,3 +638,19 @@ def function_log(self, message):
 
     def function_fail(self, message=''):
         self._run(f'{self._function_cmd_prefix}-fail', f"{message}")
+
+    def network_get(self, endpoint_name, relation_id=None):
+        """Return network info provided by network-get for a given endpoint.
+
+        endpoint_name -- A name of an endpoint (relation name or extra-binding name).
+        relation_id -- An optional relation id to get network info for.
+        """
+        cmd = ['network-get', endpoint_name]
+        if relation_id is not None:
+            cmd.extend(['-r', str(relation_id)])
+        try:
+            return self._run(*cmd, return_output=True, use_json=True)
+        except ModelError as e:
+            if 'relation not found' in str(e):
+                raise RelationNotFoundError() from e
```

Comment:
Digging into error messages is a bad idea as we discussed in similar cases before.

Why is that necessary?

Reply:
In this case I am following what we agreed on in #60 for other hook tools:

https://github.com/canonical/operator/pull/60#pullrequestreview-324357752
https://github.com/canonical/operator/blame/master/ops/model.py#L536-L564

IIRC, this was a compromise as Juju does not provide a better way to indicate a particular error type.

> Why is that necessary?

To multiplex based on what happened at the controller side:

1. a generic failure at the controller side;
2. a relation with the specified ID does not exist.

Reply:
Sorry, my bad. I forgot that this was a case we already agreed on.